lib/Backend/BackwardPass.cpp

```diff
@@ -2107,6 +2107,16 @@ BackwardPass::DeadStoreTypeCheckBailOut(IR::Instr * instr)
         return;
     }
 
+    // If oldBailoutKind is equivTypeCheck and appropriate opnd doesn't have initial/final type
+    // then leave alone the bailout
+    if (oldBailOutKind == IR::BailOutFailedEquivalentTypeCheck || oldBailOutKind == IR::BailOutFailedEquivalentFixedFieldTypeCheck)
+    {
+        if (!(propertySymOpnd->HasInitialType() || propertySymOpnd->HasFinalType()))
```

Comment:
That's what I thought, but we do end up here with non-null initialType and finalType and if I don't have this, we hit assert in Lower where that if it is not equivTypeCheck then there shouldn't be equiv bailouts.
```c++
Assert(doEquivTypeCheck || !instrChk->HasEquivalentTypeCheckBailOut());
```